### ISSUE TITLE
The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.

### ISSUE THREAD

**Issue by Balkoth** [dotnet/command-line-api#2429]
Are there specific prerequisites needed to get these projects to load1 in Visual Studio? [This](https://github.com/dotnet/command-line-api/blob/main/CONTRIBUTING.md) does not mention anything special.

`error  : The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.  C:\Work\VisualStudioGitHub\command-line-api\Directory.Build.props`

There are lots of questions around the web in regard to errors with 'Microsoft.DotNet.Arcade.Sdk' but no actual solution. So what am i missing?

**Comment by Bertk** [comment]
Hi,
[Microsoft.Dotnet.Arcade.Sdk](https://github.com/dotnet/arcade) documentation is located at GitHub.

> Arcade is intended to provide well-understood and consistent mechanisms for consuming, updating, and sharing infrastructure across the .NET Core team

https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-eng/NuGet/Microsoft.DotNet.Arcade.Sdk/overview/9.0.0-beta.24260.2

Please add additional nuget package sources in Nuget Package Manager.

`https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json`

**Comment by Balkoth** [comment]
This is what it looks like after double clicking `System.CommandLine.sln` from explorer, so it is already there:
<img width="697" alt="image" src="https://github.com/dotnet/command-line-api/assets/80055480/386e7f19-f2bd-4b29-9633-fb3b86931ffb">

**Comment by Bertk** [comment]
I created a fork and was able to build the solution with Visual Studio 2022 without errors.

<img width="203" alt="image" src="https://github.com/dotnet/command-line-api/assets/2315215/dac5a654-c37c-4fd3-a201-41329c5f0c16">


I manually executed the build for the EndToEndTestApp `dotnet build dotnet-suggest.Tests.csproj`

<img width="244" alt="image" src="https://github.com/dotnet/command-line-api/assets/2315215/345a06ae-5b24-493a-a555-299b8a561aaf">


Please check the .NET runtimes on your desktop. Maybe one is missing e.g. **dotnet 5.0.17**

```json
{
  "tools": {
    "dotnet": "8.0.101",
    "runtimes": {
      "dotnet": [
        "5.0.17",
        "6.0.7",
        "7.0.9"
      ]
    }
  },
  "msbuild-sdks": {
    "Microsoft.DotNet.Arcade.Sdk": "8.0.0-beta.24170.6"
  }
}
```

**Comment by Balkoth** [comment]
No, this does not matter. What matters on my dev machine is that i start Visual Studio with an admin account. Then it suddenly can open those projects and does not show this error.

I am still clueless what is causing this problem.

**Comment by Balkoth** [comment]
I am using the official [contribution](https://github.com/dotnet/command-line-api/blob/main/CONTRIBUTING.md) documentation, there is no such info to run Visual Studio as administrator.

And btw you are mixing running command line commands with Visual Studio. The original issue is that the projects on my machine do not load because of the error outlined in the op.

**Comment by jonsequitur** [comment]
I believe there's a package missing from the preview feed. You can potentially bypass this by using a release version of the SDK, by setting an environment variable `DisableArcade` to `1`.

**Comment by Balkoth** [comment]
Two things need to be done in order to successfully load and build the projects:

- Set environment variable `DisableArcade` to `1`
- Delete `NuGet.config` from the root directory

**Comment by Bertk** [comment]
> And btw you are mixing running command line commands with Visual Studio. The original issue is that the projects on my machine do not load because of the error outlined in the op.

The VS behavior is different from the shell scripts and the main usage for arcade is the CI based using shell scripts. The VS behavior is narrowed.

**Comment by Balkoth** [comment]
I believe we can agree that the goal should be:

- Read and follow the steps in [CONTRIBUTING.md](https://github.com/dotnet/command-line-api/blob/main/CONTRIBUTING.md)
- Be able to open and build the solution in Visual Studio

If it's not that way everyone's time trying to investigate this project is wasted with first trying to figure out to get started...

**Comment by BenjaminMichaelis** [comment]
We are interested in finding and solving any issues in the [CONTRIBUTING.md file](https://github.com/dotnet/command-line-api/blob/main/CONTRIBUTING.md) that there are and really want to. I attempted to reproduce the issue on a fresh machine, but I was unable to do so. This may point to a machine specific issue, that we want to identify and note in the [CONTRIBUTING.md file](https://github.com/dotnet/command-line-api/blob/main/CONTRIBUTING.md) as needed, so any more information you may have would be helpful..

**Comment by Balkoth** [comment]
Tell me what you need to know and i will try to provide this information.

**Comment by voroninp** [comment]
@Balkoth , for me also worked this:
```
  <packageSourceMapping>
    <clear/>
  </packageSourceMapping>
```

**Comment by xortock** [comment]
For those who would like the resolve this issue without any changes to the repo. Removing the package source mappings as shown in the image below also works

![Image](https://github.com/user-attachments/assets/b40d7edf-ec93-4afc-8c29-a554da850260)

**Comment by Balkoth** [comment]
Be carefull, you mess up your environment when you clear this in Visual Studio.

**Comment by xetle** [comment]
@Balkoth @voroninp 
> ```
>   <packageSourceMapping>
>     <clear/>
>   </packageSourceMapping>
> ```

Just adding this also worked for me. (I do not have local admin rights)
